packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -640,5 +689,15 @@ class _OngoingSubtask {
 
   final List<int> partBodyChunk;
   final int partNumber;
-  final Future<void> request;
+  Future<void> request;
```

Comment:
I would change back to `final`/`const` and add a `copyWith` method. It will make it easier to migrate this to a state machine later on.